packages/sdk/src/client/sam/contract.ts

```diff
@@ -230,16 +231,41 @@ export async function SamTotalBuyPrice(
 export async function SamEditionInfo(
   this: SoundClientInstance,
   { editionAddress, assumeValidSoundContract = false }: SamEditionAddress,
-) {
+): Promise<ExpandTypeChainStructOutput<V1_1SAMInfoStructOutput> | null> {
   const samAddress = await SamContractAddress.call(this, { editionAddress, assumeValidSoundContract })
-
   if (!samAddress) return null
 
-  const { providerOrSigner } = await this.expectProviderOrSigner()
+  const {
+    expectProviderOrSigner,
+    instance: { idempotentCachedCall },
+  } = this
 
-  const samContract = SAM__factory.connect(samAddress, providerOrSigner)
+  const { providerOrSigner } = await expectProviderOrSigner()
 
-  const info: ExpandTypeChainStructOutput<SAMInfoStructOutput> = await samContract.samInfo(editionAddress)
+  const interfaceId = await idempotentCachedCall(`minter-interface-id-${samAddress}`, async () => {
+    const samModule = SAM__factory.connect(samAddress, providerOrSigner)
+    return (await samModule.moduleInterfaceId()) as SAMInterfaceId
+  })
 
-  return { ...info }
+  switch (interfaceId) {
+    case interfaceIds.ISAM: {
+      const samContract = SAM__factory.connect(samAddress, providerOrSigner)
+
+      const [info, platformFeeBPS] = await Promise.all([
+        samContract.samInfo(editionAddress),
```

Comment:
with viem refactor, this will be easier to consolidate into 1 network request

currently, this read function is not used in our FE, we index the data in our BE when an event comes in and serve from our api in SamConfig